### DISCUSSION TITLE
QueryablePropertyAttribute

### DISCUSSION THREAD

**Discussion by Makajda** [CommunityToolkit/Maui#544] [New Feature Discussions]
Hello!

Similar to ObservablePropertyAttribute, would it make sense to add a QueryablePropertyAttribute?

[QueryableProperty] object myProperty;

It generate just like ObservableProperty + [QueryProperty(nameof(MyProperty), nameof(MyProperty))] for a class

Thank you.

**Comment by bijington** [comment]
There is already a `[QueryProperty]` attribute in .NET MAUI. https://docs.microsoft.com/en-us/dotnet/maui/fundamentals/shell/navigation#process-navigation-data-using-query-property-attributes


It doesn't generate the property for you though

**Reply by Makajda** [reply]
Yes. You need to add both an attribute and a property. QueryableProperty will add the attribute for us.

**Reply by bijington** [reply]
My main hesitation is that I don't think we want to be generating the properties especially when the MVVM toolkit already handles the observable property generation. I am keen to hear what others think though

**Reply by pictos** [reply]
I agree with @bijington. You should raise this on the .net Maui repo, I believe that's the best place to implement this enhancement.

**Comment by CliffAgius** [comment]
Closed as answered
